seed/models/building_file.py

```diff
@@ -227,13 +229,43 @@ def process(self, organization_id, cycle, property_state=None, property_id=None)
 
             scenario.save()
 
-        PropertyAuditLog.objects.create(
-            organization_id=organization_id,
-            state_id=self.property_state_id,
-            name='Import Creation',
-            description='Creation from Import file.',
-            import_filename=self.file.path,
-            record_type=AUDIT_IMPORT
-        )
+        if property_view:
+            # create a new blank state to merge the two together
+            merged_state = PropertyState.objects.create(organization_id=organization_id)
+
+            # assume the same cycle id as the former state.
+            # should merge_state also copy/move over the relationships?
+            merged_state, changed = merge_state(merged_state, property_view.state, property_state,
+                                                get_state_attrs([property_view.state, property_state]))
+
+            # log the merge
+            # Not a fan of the parent1/parent2 logic here, seems error prone, what this
+            # is also in here: https://github.com/SEED-platform/seed/blob/63536e99cf5be3a9a86391c5cead6dd4ff74462b/seed/data_importer/tasks.py#L1549
+            PropertyAuditLog.objects.create(
+                organization_id=organization_id,
+                parent1=PropertyAuditLog.objects.filter(state=property_view.state).first(),
+                parent2=PropertyAuditLog.objects.filter(state=property_state).first(),
+                parent_state1=property_view.state,
+                parent_state2=property_state,
+                state=merged_state,
+                name='System Match',
+                description='Automatic Merge',
+                import_filename=None,
+                record_type=AUDIT_IMPORT
+            )
+
+            property_view.state = merged_state
+            property_view.save()
+
+            merged_state.merge_state = MERGE_STATE_MERGED
+            merged_state.save()
```

Comment:
Wow, the different uses of the word state here had me thrown.  I get it though.  You merge two property states together, and store the newly merged state onto the property view's state member, and on the merged state's merge_state member, you store the status merge.